### PR TITLE
website: updates for the Terraform Cloud docs split

### PR DESCRIPTION
**Pull request by nfagerlund**
The Terraform Enterprise brand has now been split into two parts:

- Terraform Cloud is the application that helps teams use Terraform together,
  with remote state storage, a shared run environment, etc.
- Terraform Enterprise is the on-premise distribution that lets enterprises run
  a private instance of the Terraform Cloud application.

The former TFE docs have been split accordingly.

This PR also catches two other incidental bits: 

- Update the example command list for 0.12, bc we missed that.
- Basically delete the terraform push docs, because there's nothing for it to push to anymore.

### CODE REVIEW

File: website/docs/backends/types/remote.html.md [hunk 1]
```diff
@@ -11,20 +11,20 @@ description: |-
...
+[app.terraform.io](https://app.terraform.io) or a Terraform Enterprise instance
+(version v201809-1 or newer).
 
-The remote backend stores state and runs operations in Terraform Enterprise.
+The remote backend stores state and runs operations in Terraform Cloud.
 
-When used with a Pro or Premium tier Terraform Enterprise account, operations
+When used with a Pro or Premium tier Terraform Cloud account, operations
 like `terraform plan` or `terraform apply` are executed in Terraform
 Enterprise's run environment, with log output streaming to the local terminal.
```
Comment (mildwonkey) [inline website/docs/backends/types/remote.html.md]: should the next two `Enterprise`s also be changed to `Cloud`?

Comment (nfagerlund) [reply website/docs/backends/types/remote.html.md]: 😱 They should, but I missed them because HARD WRAP 😭🤬

Excellent example of why I hate hard wrap.

Comment (nfagerlund) [reply website/docs/backends/types/remote.html.md]: Fixed!
